cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/BasicMapReduceTaskContext.java

```diff
@@ -192,8 +208,38 @@ public LoggingContext getLoggingContext() {
     return loggingContext;
   }
 
+  //---- following are methods to manage transaction lifecycle for the datasets. This needs to
+  //---- be refactored after [TEPHRA-99] and [CDAP-3893] are resolved.
+
+  private void initializeTransactionAwares() {
```

Comment:
right. And there was a small bug there, too: We do not start a transaction context in the dataset cache (because we have to reuse an existing tx), hence getTransactionAwares() would return nothing.  Fixed that now by adding an extra method that returns all static tx-awares regardless of whether a tx context was created.